lib/transport/transport-factory-id.h

```diff
@@ -34,14 +34,12 @@ void transport_factory_id_global_deinit(void);
 void transport_factory_id_register(TransportFactoryId *);
 GList *transport_factory_id_clone_registered_ids(void);
 void transport_factory_id_free(gpointer);
-gpointer transport_factory_id_clone(gconstpointer);
 const gchar *transport_factory_id_to_string(const TransportFactoryId *);
+guint transport_factory_id_hash(gconstpointer);
+gboolean transport_factory_id_equal(const TransportFactoryId *, const TransportFactoryId *);
+TransportFactoryId *transport_factory_id_clone(const TransportFactoryId *);
 
 #define TRANSPORT_FACTORY_ID_NEW(name) ({GString *str = g_string_new(""); g_string_printf(str, "%s-%s:%d", name, __FILE__, __LINE__); str;})
-#define TRANSPORT_FACTORY_ID_FREE_FUNC transport_factory_id_free
-#define TRANSPORT_FACTORY_ID_HASH_FUNC g_string_hash
-#define TRANSPORT_FACTORY_ID_CMP_FUNC g_string_equal
-#define TRANSPORT_FACTORY_ID_CLONE transport_factory_id_clone
```

Comment:
I'd also squash this commit into `transport: add transport-factory-id` and use the new non-macro interface in `transport: add TransportFactoryRegistry`.